### PR TITLE
Minor fixes for 2.1 release

### PR DESCRIPTION
**Pull request by SaswatPadhi**
Mostly spacing issues and missing / misplaces parentheses.

One question @ajreynol @polgreen:
Is an IO oracle supposed to be the same as a positive witness oracle? If so, is there a reason to keep both?

### CODE REVIEW

**Comment by ajreynol** [comment]
The changes look good to me but I'd like @polgreen to also have a look.

Regarding io vs. positive witness, I agree this is fairly redundant. Maybe @polgreen has an opinion about whether these are necessary?  

It seems this was made to be symmetric to the negative witness case. I'd suggest we at least merge the explanation of the two commands, since they reduce to the same thing.

**Comment by polgreen** [comment]
Sorry, I don't know how I missed this. input-output and positive witness oracles should be different: 
- An io oracle receives a specific input and returns the correct output for that input.
-  A positive witness has no input parameters, and returns some correct input-output pair, so it can return any input-output pair and the user has no control over which one it returns.

But i might have made a mistake when writing them out, let me take a look

File: language/sygus.tex [hunk 1]
```diff
@@ -1290,116 +1284,95 @@ \subsection{Declaring Oracle Functional Symbols}
...
+It is syntactic sugar for:
 
-$\paren{\constraintoraclekwd\mbox{ } 
-\mbox{ }
-\paren{\paren{x_1\mbox{ }\sigma_1} \ldots \paren{x_n\mbox{ }\sigma_n} \paren{x \mbox{ }\sigma}}\mbox{ }
-\paren{{\tt=}\mbox{ } \paren{F \mbox{ } x_1 \ldots x_n}\mbox{ }  x}\mbox{ }N
-}
-$
+\begin{lstlisting}[language=SyGuS-Desugar]
+($\constraintoraclekwd$ (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)) (($x$ $\sigma$)) (= ($F$ $x_1$ $\ldots$ $x_n$) $x$) $N$)
```
Comment (polgreen) [inline language/sygus.tex]:
```suggestion
($\constraintoraclekwd$ () (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)($x$ $\sigma$)) (= ($F$ $x_1$ $\ldots$ $x_n$) $x$) $N$)
```

File: language/sygus.tex [hunk 2]
```diff
@@ -1290,116 +1284,95 @@ \subsection{Declaring Oracle Functional Symbols}
...
-$\paren{\constraintoraclekwd\mbox{ } 
-\mbox{ }
-\paren{\paren{x_1\mbox{ }\sigma_1} \ldots \paren{x_n\mbox{ }\sigma_n} \paren{x \mbox{ }\sigma}}\mbox{ }
-\paren{{\tt not} \mbox{ } \paren{{\tt=} \mbox{ } \paren{F \mbox{ } x_1 \ldots x_n}\mbox{ }  x}} \mbox{ } N}
-$
+This command declares a negative witness oracle for synthesis function $F$.
+It is syntactic sugar for:
 
+\begin{lstlisting}[language=SyGuS-Desugar]
+($\constraintoraclekwd$ (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)) (($x$ $\sigma$)) (not (= ($F$ $x_1$ $\ldots$ $x_n$) $x$)) $N$)
```
Comment (polgreen) [inline language/sygus.tex]:
```suggestion
($\constraintoraclekwd$ ()(($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)($x$ $\sigma$)) (not (= ($F$ $x_1$ $\ldots$ $x_n$) $x$)) $N$)
```

File: language/sygus.tex [hunk 3]
```diff
@@ -1290,116 +1284,95 @@ \subsection{Declaring Oracle Functional Symbols}
...
-\paren{{\tt=} \mbox{ } \paren{{\tt=}\mbox{ } \paren{F \mbox{ } x_1 \ldots x_n } \mbox{ }  x} \mbox{ } R}  \mbox{ }N}
-$
+This command declares a membership-query oracle for synthesis function $F$.
+It is syntactic sugar for:
 
-% 
+\begin{lstlisting}[language=SyGuS-Desugar]
+($\constraintoraclekwd$
+  (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$))
+  (($x$ $\sigma$) ($R$ Bool))
```
Comment (polgreen) [inline language/sygus.tex]: The parenthesis have changed here, should have
input: (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)($x$ $\sigma$))
output: (($R$ Bool))

Comment (polgreen) [reply language/sygus.tex]:
```suggestion
  (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)($x$ $\sigma$))
   (($R$ Bool))
```

**Review by polgreen — CHANGES_REQUESTED** [review]

File: language/sygus.tex (hunk 1 above)
Comment (polgreen) [reply language/sygus.tex]: this was my mistake when i wrote the original, but it should have no inputs, and all of the params are part of the output

File: language/sygus.tex (hunk 2 above)
Comment (polgreen) [reply language/sygus.tex]: ditto: should have no inputs, all of the params are outputs

**Comment by ajreynol** [comment]
I agree we should keep both then.  Let's fix the parentheses on this PR.

**Comment by SaswatPadhi** [comment]
Thanks @ajreynol, @polgreen.

I understand the difference between the two now. Let me push a change to fix this as Elizabeth suggested above.

File: language/sygus.tex (hunk 1 above)
Comment (SaswatPadhi) [reply language/sygus.tex]: Thanks, fixed as suggested.

File: language/sygus.tex (hunk 3 above)
Comment (SaswatPadhi) [reply language/sygus.tex]: Thanks, fixed as suggested.

File: language/sygus.tex (hunk 2 above)
Comment (SaswatPadhi) [reply language/sygus.tex]: Thanks, fixed as suggested.

**Comment by SaswatPadhi** [comment]
Fixed the desugared oracle constraints as suggested by @polgreen.

Please take another look.

File: language/sygus.tex [hunk 4]
```diff
@@ -1290,116 +1284,103 @@ \subsection{Declaring Oracle Functional Symbols}
...
 
-
-$\paren{\constraintoraclekwd\mbox{ } \paren{\paren{x_1\mbox{ }\sigma_1} \ldots \paren{x_n\mbox{ }\sigma_n}}\mbox{ }
-\paren{\paren{x \mbox{ }\sigma}}
-\paren{{\tt=}\mbox{ } \paren{F \mbox{ } x_1 \ldots x_n}\mbox{ }  x}\mbox{ }N
-}
-$
-
+\begin{lstlisting}[language=SyGuS-Desugar]
+($\constraintoraclekwd$ (($x_1$ $\sigma_1$) $\ldots$ ($x_n$ $\sigma_n$)) (($x$ $\sigma$)) (= ($F$ $x_1$ $\ldots$ $x_n$) $x$) $N$)
```
Comment (polgreen) [inline language/sygus.tex]: should this have the same new line pattern as all the other oracle sugar? i.e., 

(inputs on one line)
(then outputs)
(then the constraint term)
then the binary name

**Review by polgreen — APPROVED** [review]
looks fine to me now. Maybe the i-o oracle should have the same formatting pattern as the other oracle sugar commands

File: language/sygus.tex (hunk 4 above)
Comment (SaswatPadhi) [reply language/sygus.tex]: Thanks, good catch. Fixed it now

**Pull request merged**